### PR TITLE
Make test_bank_cloned_stake_delegations() aware of stake minimum delegation

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

The bank test `test_bank_cloned_stake_delegations()` fails once the stake minimum delegation is raised. See #24603 for more context.

#### Summary of Changes

- Update bank test_bank_cloned_stake_delegations() to be aware of stake minimum delegation

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -11362,13 +11362,21 @@ pub(crate) mod tests {
...
-        } = create_genesis_config_with_leader(500, &solana_sdk::pubkey::new_rand(), 1);
+        } = create_genesis_config_with_leader(10_000_000_000, &solana_sdk::pubkey::new_rand(), 1);
         let bank = Arc::new(Bank::new_for_tests(&genesis_config));
 
         let stake_delegations = bank.stakes_cache.stakes().stake_delegations().clone();
         assert_eq!(stake_delegations.len(), 1); // bootstrap validator has
                                                 // to have a stake delegation
 
+        let balance = {
+            let rent = &bank.rent_collector().rent;
```
Comment (joncinque) [inline runtime/src/bank.rs:11373]: I got had by this while updating the RPC test -- could you double-check that there's non-zero rent here? https://github.com/solana-labs/solana/pull/24670/files#diff-6507431a1db5f52ddbedfa0242280da09cca66214f3ca42e0aba4387e6b1d3e7 has the fix

**Review by joncinque — COMMENTED** [review]
Looks good to go otherwise, just the one question about rent.

File: runtime/src/bank.rs (hunk 1 above)
Comment (brooksprumo) [reply runtime/src/bank.rs:11378]: Yah, by default rent is free. Updated to use the default Rent. Done!

**Review by joncinque — APPROVED** [review]
Awesome, thanks for adding the non-free rent!

**Pull request merged**